test/e2e-olm/elasticsearch_test.go

```diff
@@ -1,318 +1,484 @@
 package e2e
 
 import (
+	"context"
 	"fmt"
 	"testing"
 	"time"
 
+	loggingv1 "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
 	"github.com/openshift/elasticsearch-operator/test/utils"
+
+	"github.com/operator-framework/operator-sdk/pkg/test"
 	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
-	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
 
-	goctx "context"
+func TestElasticsearchCluster(t *testing.T) {
+	utils.RegisterSchemes(t)
+	t.Run("Single node", singleNodeTest)
+	t.Run("Multiple nodes", multipleNodesTest)
+	t.Run("Scale up nodes", scaleUpNodesTest)
+	t.Run("Multiple nodes with a single non-data node", multipleNodesWithNonDataNodeTest)
+	t.Run("Full cluster redeploy", fullClusterRedeployTest)
+	t.Run("Rolling restart", rollingRestartTest)
+	t.Run("Invalid master count", invalidMasterCountTest)
+}
 
-	elasticsearch "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
-	framework "github.com/operator-framework/operator-sdk/pkg/test"
-	v1 "k8s.io/api/core/v1"
-)
+func singleNodeTest(t *testing.T) {
+	f := test.Global
+
+	ctx := test.NewContext(t)
```

Comment:
If we are going to split theses tests why not take the opportunty to refactor common functionality into comment functions

Reply:
The test context is part of each test case to ensure isolation and cleanup per test case.

Reply:
FWIW, I suggest to keep grouping common functionality in a separate PR that addresses refactoring e2e to gingko/gomega which address BeforeEach/AfterEach/etc. more intuitively.